### DISCUSSION TITLE
Consider using chiseled containers

### DISCUSSION THREAD

**Discussion by tlunsfordCXP** [devcontainers/community#87] [Ideas]
Microsoft just announced the production-quality release of chiseled containers. The following post gives a good summary of chiseled containers, and it seems to be in strong alignment with a devcontainer-based approach.

https://devblogs.microsoft.com/dotnet/announcing-dotnet-chiseled-containers/

I'd like to prove out use of chiseled containers in a base devcontainer, but I'm not yet experienced enough to know where to start. Is anyone here interested in working together to prove it out?

**Comment by samruddhikhandale** [comment]
Hello 👋 

Chiseled containers definitely sounds pretty intriguing due to improved security and performance. Thank you for sharing this with us! 

Are you in the #devcontainer-community slack channel? That's another way to engage with the community and share thoughts/ideas with others. See https://github.com/orgs/devcontainers/discussions/3

> Consider using chiseled containers
>
> I'd like to prove out use of chiseled containers in a base devcontainer

I can see this question going in few different directions, @tlunsfordCXP can you help me clarify my understanding? thank you! 

1. Update the Dev Containers Spec to have a new and different schema for `devcontainer.json` to support the ideology of chiseled containers (eg. rootless, no shell)
2. @devcontainers/cli should support creating chiseled dev containers by essentially using chiseled images? (eg. `mcr.microsoft.com/dotnet/aspnet:8.0-jammy-chiseled`)
3. Create minimal Docker images (aka chiseled) for the images currently hosted from https://github.com/devcontainers/images ? essentially, starting with the [base-ubuntu](https://github.com/devcontainers/images/tree/main/src/base-ubuntu) image?

I believe (1) and (2) would be needed to provide a complete and effective `Chiseled dev containers` experience.
Let me know if I have misunderstood anything.

**Reply by tlunsfordCXP** [reply]
Thank you for the response. I have now requested slack access.

I'm pretty new to devcontainers, so your questions are far ahead of me regarding direction(s) to go in. I could see all 3 of your items being of value, in the sequence you provided.
